packages/app/src/providers/h5wasm/h5wasm-api.ts

```diff
@@ -0,0 +1,153 @@
+import type {
+  ArrayShape,
+  Attribute,
+  AttributeValues,
+  Dataset,
+  Entity,
+  GroupWithChildren,
+  NumericType,
+  Shape,
+} from '@h5web/shared';
+import { hasScalarShape, EntityKind } from '@h5web/shared';
+import type { Dataset as h5wasm_Dataset } from 'h5wasm';
+import {
+  ready as h5wasm_ready,
+  File,
+  Group as h5wasm_Group,
+  FS as h5wasm_FS,
+} from 'h5wasm';
+
+import { ProviderApi } from '../api';
+import type { ExportFormat, ValuesStoreParams } from '../models';
+import { convertDtype, handleAxiosError } from '../utils';
+import type {
+  H5WasmAttribute,
+  H5WasmAttrValuesResponse,
+  H5WasmDataResponse,
+  H5WasmEntityResponse,
+} from './models';
+
+function convert_attrs(attrs: object): Attribute[] {
+  return Object.entries(attrs).map(([name, value]) => ({
+    name,
+    shape: value.shape,
+    type: convertDtype((value as h5wasm_attr).dtype),
+  }));
+}
+
+interface h5wasm_attr {
+  value: unknown;
+  shape: Shape;
+  dtype: string;
+}
+
+export class H5WasmApi extends ProviderApi {
+  /* API compatible with h5wasm@0.2.10 */
+
+  public file_promise: Promise<File>;
+
+  public constructor(filepath: string) {
+    super(filepath);
+    this.file_promise = this.fetchFile(filepath);
+  }
+
+  public async getEntity(path: string): Promise<Entity> {
+    const file = await this.file_promise;
+    const entity_obj = file.get(path);
+    return this.processEntityObject(path, path, entity_obj, true);
+  }
+
+  public async getValue(
+    params: ValuesStoreParams
+  ): Promise<H5WasmDataResponse> {
+    const { dataset } = params;
+    const file = await this.file_promise;
+    const dataset_obj = file.get(dataset.path) as h5wasm_Dataset;
```

Comment:
Sure - that sounds fine.  I removed the ```process...``` functions because most of them are trivial in this case, but keeping a structure that is similar to other providers is important.

Reply:
Sure, if the `process...` is trivial and can be removed, that is even better. , [Same for `...Response` types](https://github.com/silx-kit/h5web/pull/1031#discussion_r838442734) if you can use directly `h5wasm` classes.

To me, the important part is that we make clear in the `api` what is the format/shape of the objects returned by `h5wasm`. In other words, have the return type of `fetch...` functions match the response. Then, the TypeScript processing will flow naturally.